### PR TITLE
ref(discover): map project to project.name in discover

### PR DESCRIPTION
**Pull request by DominikB2014**
The frontend field renderer expects the field to be called `project`, https://github.com/getsentry/sentry/blob/5eed07ecb584cf297ceacd5697bc6a2469908343/static/app/utils/discover/fieldRenderers.tsx#L507

but this isn't a span_metrics field so currently we query for `project.id`, but this results in us duplicating table cells like here
https://github.com/getsentry/sentry/blob/91712ae955a0dd1c16f16f08c516389570b033d7/static/app/views/performance/http/projectIdCell.tsx#L9 instead of relying on the existing field renderers.

We can avoid this entirely by remapping `project` => `project.slug`

@wmak any side effects from this that you can think of?

### CODE REVIEW

File: tests/snuba/api/endpoints/test_organization_events_span_metrics.py [hunk 1]
```diff
@@ -1479,6 +1479,52 @@ def test_count_op(self):
...
+            }
+        )
+
+        assert response.status_code == 200, response.content
+        assert response_2.status_code == 200, response_2.content
+        data = response.data["data"]
+        data_2 = response_2.data["data"]
+
+        assert len(data) == len(data_2)
+        assert data[0]["project.id"] == data_2[0]["project"]
```
Comment (wmak) [inline tests/snuba/api/endpoints/test_organization_events_span_metrics.py:1525]: nit: we should compare to the known test data and not the responses against each other

File: src/sentry/search/events/datasets/spans_metrics.py [hunk 2]
```diff
@@ -43,6 +43,12 @@ def field_alias_converter(self) -> Mapping[str, Callable[[str], SelectType]]:
             constants.DEVICE_CLASS_ALIAS: lambda alias: field_aliases.resolve_device_class(
                 self.builder, alias
             ),
+            constants.PROJECT_ALIAS: lambda alias: field_aliases.resolve_project_slug_alias(
+                self.builder, alias
+            ),
+            constants.PROJECT_SLUG_ALIAS: lambda alias: field_aliases.resolve_project_slug_alias(
+                self.builder, alias
+            ),
```
Comment (DominikB2014) [inline src/sentry/search/events/datasets/spans_metrics.py:51]: Any issues duplicating this here? Or better to map "project" -> "project.slug" and use one alias function

Comment (wmak) [reply src/sentry/search/events/datasets/spans_metrics.py:51]: Nah this is completely fine :+1:

File: src/sentry/search/events/datasets/spans_metrics.py [hunk 3]
```diff
@@ -43,6 +43,12 @@ def field_alias_converter(self) -> Mapping[str, Callable[[str], SelectType]]:
             constants.DEVICE_CLASS_ALIAS: lambda alias: field_aliases.resolve_device_class(
                 self.builder, alias
             ),
+            constants.PROJECT_ALIAS: lambda alias: field_aliases.resolve_project_slug_alias(
+                self.builder, alias
+            ),
+            constants.PROJECT_SLUG_ALIAS: lambda alias: field_aliases.resolve_project_slug_alias(
```
Comment (wmak) [inline src/sentry/search/events/datasets/spans_metrics.py:49]:
```suggestion
            constants.PROJECT_NAME_ALIAS: lambda alias: field_aliases.resolve_project_slug_alias(
```

**Review by wmak — APPROVED** [review]
I told you the wrong alias in slack sorry 🙃, otherwise LGTM

**Pull request merged**